server/routes/api.go

```diff
@@ -35,20 +36,33 @@ import (
 
 	"github.com/temporalio/ui-server/server/config"
 	"github.com/temporalio/ui-server/server/generated/api/workflowservice/v1"
+	"github.com/temporalio/ui-server/server/rpc"
 )
 
 // SetAPIRoutes sets api routes
-func SetAPIRoutes(e *echo.Echo, cfg *config.Config, temporalConn *grpc.ClientConn) error {
+func SetAPIRoutes(e *echo.Echo, cfgProvider *config.ConfigProviderWithRefresh) error {
 	api := e.Group("/api")
 	api.GET("/v1/me", getCurrentUser)
-	api.GET("/v1/settings", getSettings(cfg))
-	api.Match([]string{"GET", "POST", "PUT", "PATCH", "DELETE"}, "/*", temporalAPIHandler(temporalConn))
+	api.GET("/v1/settings", getSettings(cfgProvider))
+	api.Match([]string{"GET", "POST", "PUT", "PATCH", "DELETE"}, "/*", temporalAPIHandler(cfgProvider))
 	return nil
 }
 
-func temporalAPIHandler(temporalConn *grpc.ClientConn) echo.HandlerFunc {
+func temporalAPIHandler(cfgProvider *config.ConfigProviderWithRefresh) echo.HandlerFunc {
 	return func(c echo.Context) error {
-		mux, err := getTemporalClientMux(c, temporalConn)
+		cfg, err := cfgProvider.GetConfig()
+		if err != nil {
+			return c.JSON(http.StatusInternalServerError, err)
+		}
+
+		tls, err := rpc.CreateTLSConfig(cfg.TemporalGRPCAddress, &cfg.TLS)
```

Comment:
Is this called on every request and never cached?

Reply:
it's called on every request though receives a cached config https://github.com/temporalio/ui-server/pull/102/files#diff-0e504256c68c0d5b7cd3c799aa6c942469504ebb007e337f33cef429241061a1R67

Reply:
Config passed into `rpc.CreateTLSConfig()` is cashed, but the output of it is generated every time? And then `rpc.CreateGRPCConnection()` is also called every time. Shouldn't they be cached?

Reply:
right, figured to cached the GRPC connections once we add multi-cluster support and work with the pool of connections. Can add caching now